.github/workflows/test.yml

```diff
@@ -164,7 +164,7 @@ jobs:
           key: ${{ runner.os }}-hf-model-${{ env.model_key }}
 
       - name: "Download HF models"
-        if: steps.changed-src-files.outputs.any_changed == 'true' && steps.cache_restore.outputs.cache-hit != 'true'
+        if: steps.changed-src-files.outputs.any_changed == 'true'
```

Comment:
This was a "proxy" condition to not run this step if no sources were changed, i.e. for documentation only updates.

The same awkward condition got sprinkled all over the test workflow to keep the test workflow required for all PRs (and run it all the time), yet not actually execute the logic inside the workflow. All other options seemed worse.

Reply:
If we want to keeping the same logic to run all steps yet not execute the logic inside the steps, we probably could word the condition a bit more succinctly.

```yaml
if: steps.changed-src-files.outputs.any_changed == 'true'
```

```yaml
if: env.SKIP_TESTS == 'true'
```

Reply:
Should we be keeping the `&& steps.cache_restore.outputs.cache-hit != 'true'` condition around though? We don't want to run the download step if the cache hit, right?